client/lib/menu-data/menu-data.js

```diff
@@ -367,27 +386,31 @@ MenuData.prototype.interceptLoadForHomepageLink = function( menu ) {
 	return menu;
 };
 
-MenuData.prototype.saveMenu = function( menu, callback ) {
-	if ( ! menu ) {
-		menu = this.find( { id: this.lastChangedMenuID } );
-	}
-
-	if ( ! this.isValidMenu( menu ) ) {
-		callback && callback( new Error( 'Invalid menu' ) );
-		debug( 'saveMenu: fail' );
-		return;
-	}
-
-	menu = cloneDeep( menu );
-
-	if ( menu.id === this.getDefaultMenuId() ) {
-		return this.saveDefaultMenu();
+MenuData.prototype.createNewPagePromise = ( menuItem, siteID ) => new Promise(
+	( resolve, reject ) => {
+		postEditStore.startEditingNew( { ID: siteID } );
+		postEditStore.saveEdited(
+			{
+				title: menuItem.name,
+				ID: 'new',
+				type: 'page',
+				status: 'publish'
+			},
+			function( error, data ) {
```

Comment:
Style consistency, I'd say, and an easier ability to use `this` inside the callback without having to worry about binding (or that darned `self` var).